addons/mail/tests/common.py

```diff
@@ -1951,6 +1951,16 @@ def _activate_multi_lang(cls, lang_code='es_ES', layout_arch_db=None, test_recor
             }
         })
 
+    def _filter_channels_fields(self, /, *channels_data):
+        """ Remove store channel data dependant on other modules if they are not not installed.
+        Not written in a modular way to avoid complex override for a simple test tool.
+        """
+        for data in channels_data:
+            # if 'ai_livechat' module is not installed
+            if not ('ai.agent' in self.env and 'im_livechat.channel' in self.env):
+                data.pop("livechat_with_ai_agent", None)
```

Comment:
This is still not the correct way to do it, as both livechat and ai can be installed and ai_livechat uninstalled.

We really need to check either the module adding the field specifically, or the existence of the field itself.
Maybe just `"livechat_with_ai_agent" in self.env["discuss.channel"]._fields`.